param/_utils.py

```diff
@@ -518,26 +518,52 @@ def _is_number(obj):
     else: return False
 
 
-def _is_abstract(class_):
+def _is_abstract(class_: type) -> bool:
+    if inspect.isabstract(class_):
```

Comment:
(I haven't looked where this is used; if it is not in many places, ignore me).

Is there a penalty related to this check? E.g. if this is in a hotpath will this add significant overhead?

Reply:
Timed it and it's 2-3x slower than the try/except that follows. That's also not a particularly hot path I think.